File: src/main/java/cc/sukazyo/cono/morny/bot/api/InlineQueryUnit.java
package cc.sukazyo.cono.morny.bot.api;

import com.pengrad.telegrambot.model.request.InlineQueryResult;

public class InlineQueryUnit<T extends InlineQueryResult<T>> {
	
	public static final int DEFAULT_INLINE_CACHE_TIME = 300;
	public static final boolean DEFAULT_INLINE_PERSONAL_RESP = false;
	
	private int cacheTime = DEFAULT_INLINE_CACHE_TIME;
	private boolean isPersonal = DEFAULT_INLINE_PERSONAL_RESP;
	public final T result;
	
	public InlineQueryUnit (T result) {
		this.result = result;
	}
	
	public int cacheTime () {
		return cacheTime;
	}
	
	public InlineQueryUnit<T> cacheTime (int cacheTime) {
		this.cacheTime = cacheTime;
		return this;
	}
	
	public boolean isPersonal () {
		return isPersonal;
	}
	
	public InlineQueryUnit<T> isPersonal (boolean isPersonal) {
		this.isPersonal = isPersonal;
		return this;
	}
	
}

File: src/main/scala/cc/sukazyo/cono/morny/bot/query/InlineQueryUnit.scala
package cc.sukazyo.cono.morny.bot.query

import cc.sukazyo.cono.morny.bot.query.InlineQueryUnit.defaults
import com.pengrad.telegrambot.model.request.InlineQueryResult

object InlineQueryUnit {
	
	object defaults:
		val CACHE_TIME = 1
		val IS_PERSONAL = false
	
}

class InlineQueryUnit[T <: InlineQueryResult[T]](val result: T) {
	
	var cacheTime: Int = defaults.CACHE_TIME
	var isPersonal: Boolean = defaults.IS_PERSONAL
	
	def cacheTime (v: Int): InlineQueryUnit[T] =
		this.cacheTime = v
		this
	
	def isPersonal (v: Boolean): InlineQueryUnit[T] =
		this.isPersonal = v
		this
	
}
